src/Test/Verilog/Expression.idr

```diff
@@ -0,0 +1,192 @@
+module Test.Verilog.Expression
+
+import Data.Fuel
+import Data.Vect
+
+import public Data.So
+
+import Test.DepTyCheck.Gen
+import Test.DepTyCheck.Gen.Coverage
+
+import public Test.Verilog.Module
+
+import public Test.Verilog.Literal
+
+
+%default total
+
+||| IEEE 1800-2023, 11.2
+||| An expression is a construct that combines operands with operators to produce a result that is a function of
+||| the values of the operands and the semantic meaning of the operator. Any legal operand, such as a net bit-
+||| select, without any operator is considered an expression. Wherever a value is needed in a SystemVerilog
+||| statement, an expression can be used.
+||| An operand can be one of the following:
+||| — Constant literal number, including real literals
+||| — String literal
+||| — Parameter, including local and specify parameters
+||| — Parameter bit-select or part-select, including local and specify parameters
+||| — Net (see 6.7)
+||| — Net bit-select or part-select
+||| — Variable (see 6.8)
+||| — Variable bit-select or part-select
+||| — Structure, either packed or unpacked
+||| — Structure member
+||| — Packed structure bit-select or part-select
+||| — Union, packed, unpacked, or tagged
+||| — Union member
+||| — Packed union bit-select or part-select
+||| — Array, either packed or unpacked
+||| — Packed array bit-select, part-select, element, or slice
+||| — Unpacked array element bit-select or part-select, element, or slice
+||| — A call to a user-defined function, system function, or method that returns any of the above
+public export
+data SVExpression : (expressionType : SVType) -> (ports : PortsList) -> (usedPorts : ListOfPortsIndices ports) -> Type
```

Comment:
Consider using Fin lists instead of ListOfPortsIndices for better support of pretty printers.